### ISSUE TITLE
Better support for remote apis

### ISSUE THREAD

**Issue by vincentjames501** [CuppaLabs/angular2-multiselect-dropdown#206]
Similar to #148 , it would be nice to override the default search behavior. Sometimes simple substring matching over all the objects properties isn't great or one would want to load data from a remote endpoint instead. 

```html
(onSearch)="performSearch($event)"
```

Sometimes, remote searching can take a few seconds or even longer. It would be nice to allow showing a simple loading message while hiding old results:

```html
[isLoading]="true"
[loadingText]="'Loading...'"
```

Also, sometimes data can fail to load or something else such as no results, search failure, etc. It would be nice to expose something similar for errors:

```html
[isFailure]="true"
[failureText]="'Loading...'"
```

Similar to lazy loading, it may be nice to be able to _fetch the next page of results_ from a remote server when the user hits the bottom and show the loading indicator:

```html
[hasNextPage]="true"
(onNextPage)="getNextPage($event)"
```

Ideally one could pass `[data]` as an object instead of an array of `ListItem` like:

```javascript
{
data: ListItem[],
next: any // if it's defined, it has a next page. Could be a url or cursor to the next results page
}
```

**Comment by CuppaLabs** [comment]
feature available in v2.9.0

**Comment by CuppaLabs** [comment]
refer example https://cuppalabs.github.io/angular2-multiselect-dropdown/#/customSearchAPI
